### PR TITLE
Compute error indicators after each subinterval

### LINKED ISSUE #209: Compute error indicators after each subinterval
Related to discussion in #73, we decided to compute error indicators after each subinterval rather than at the end of the entire time interval.

### PR DESCRIPTION
**Pull request by ddundo**
Closes #209.

Similarly to `MeshSeq._solve_forward` and `MeshSeq.solve_forward`, this PR makes `AdjointMeshSeq.solve_adjoint` use a generator function `AdjointMeshSeq._solve_adjoint` that `yield`s after each subinterval solve. That way we can compute error indicators in `GoalOrientedMeshSeq.indicate_errors` after each subinterval.

### CODE REVIEW

File: goalie/adjoint.py [hunk 1]
```diff
@@ -402,8 +403,8 @@ def solve_adjoint(
         :type get_adj_values: :class:`bool`
         :kwarg test_checkpoint_qoi: solve over the final subinterval when checkpointing
             so that the QoI value can be checked across runs
-        :returns: the solution data of the forward and adjoint solves
-        :rtype: :class:`~.AdjointSolutionData`
+        :yield: the solution data of the forward and adjoint solves
```
Comment (ddundo) [inline goalie/adjoint.py:406]: This should be `yields`. Will fix after review :)

**Comment by ddundo** [comment]
@jwallwork23 could you please review (once you're back from the conference - no rush)? This passes locally for me with the outdated petsc version, so I think it's ready for review. But I won't merge before https://github.com/mesh-adaptation/animate/pull/142 is merged

**Review by jwallwork23 — APPROVED** [review]
Seems sensible, thanks! I'm hopeful that the Docker build will be fixed shortly and we can then merge those changes in here to run the CI.

**Comment by jwallwork23** [comment]
I guess this needs to be rebased on top of `main` or have `main` merged into it to pick up the fix from #214.

**Pull request merged**